app/src/androidTest/java/ch/epfl/cs311/wanderwave/utils/BeaconPlacerTest.kt

```diff
@@ -0,0 +1,158 @@
+package ch.epfl.cs311.wanderwave.utils
+
+import android.util.Log
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.data.Location
+import ch.epfl.cs311.wanderwave.ui.components.utils.computeDistanceBetweenBeacons
+import ch.epfl.cs311.wanderwave.ui.components.utils.findNearbyBeacons
+import ch.epfl.cs311.wanderwave.ui.components.utils.findRandomBeacon
+import ch.epfl.cs311.wanderwave.ui.components.utils.haversine
+import ch.epfl.cs311.wanderwave.ui.components.utils.placeBeaconsRandomly
+import ch.epfl.cs311.wanderwave.ui.components.utils.randomLatLongFromPosition
+import com.google.android.gms.maps.model.LatLng
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+
+private const val BEACON_COUNT = 20
+
+@RunWith(AndroidJUnit4::class)
```

Comment:
Do you need this ? It seems like a unit test so why running it with emulator like this ?

Reply:
so actually for some really really strange reasons, when i remove it, the test from the trackListScreen where not passing (the two class are totally unrelated), so for now i think we should keep it